lib/client/src/client.rs

```diff
@@ -1757,6 +1758,21 @@ impl MangoClient {
         .send_and_confirm(&self.client)
         .await
     }
+
+    pub async fn simulate(
+        &self,
+        instructions: Vec<Instruction>,
+    ) -> anyhow::Result<RpcResult<RpcSimulateTransactionResult>> {
```

Comment:
This return type feels awkward. I'd either
- collapse to `anyhow::Result<()>` - but then it's hard to disentangle the simulation errors from other errors, and it might be cool to get post-simulation state returned
- collapse to `anyhow::Result<RpcSimulateTransactionResult>`